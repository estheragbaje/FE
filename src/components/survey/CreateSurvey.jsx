import React, { Component } from 'react';
import styled from 'styled-components';
import Divider from '../../styles/Divider';

const Question = styled.div`
  /* border: 1px solid green; */
`;

const Questions = styled.div`
  /* border: 1px solid blue; */
`;

const AddButton = styled.button`
  font-size: 1.5rem;
  border-radius: 50%;
  padding: 0px;
  width: 4rem;
  height: 4rem;
  text-align: center;
`;

class CreateSurvey extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: {
        title: '',
        questions: [],
      },
    };
  }

  render() {
    const { data } = this.state;
    return (
      <div className="container">
        <div className="col-md">
          <form className="border border-light p-5" action="#!">
            <h1 className="text-center">Create a Survey</h1>
            <Divider size={30} />
            <div className="form-group">
              <label htmlFor="title">Survey Title</label>
              <input
                type="text"
                id="title"
                className="form-control mb-4"
                value={data.title}
                onChange={this.change}
              />
            </div>

            <Questions>
              <h2>Survey Questions</h2>
              <Divider size={30} />
              <Question>
                <div className="form-group">
                  <label htmlFor="text">Question</label>
                  <textarea
                    className="form-control rounded-0"
                    id="text"
                    rows="3"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="type">Question type</label>
                  <select
                    className="browser-default custom-select mb-4"
                    id="type"
                  >
                    <option value="" disabled>
                      Choose option
                    </option>
                    <option value="text">Text</option>
                    <option value="rating">Rating</option>
                  </select>
                </div>
              </Question>

              <div className="text-center">
                <AddButton type="button" className="btn btn-info">
                  +
                </AddButton>
              </div>
            </Questions>

            <Divider size={30} />

            <div className="form-group">
              <button className="btn btn-primary btn-block" type="submit">
                Save Survey
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

export default CreateSurvey;
