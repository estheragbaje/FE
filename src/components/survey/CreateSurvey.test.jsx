import React from 'react';
import { shallow } from 'enzyme';

import CreateSurvey from './CreateSurvey';

describe('<CreateSurvey />', () => {
  it('renders the CreateSurvey component correctly', () => {
    const wrapper = shallow(<CreateSurvey />);
    expect(wrapper).toMatchSnapshot();
  });
});
