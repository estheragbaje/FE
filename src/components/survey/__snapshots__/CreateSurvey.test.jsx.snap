// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateSurvey /> renders the CreateSurvey component correctly 1`] = `
<styled.div
  className="container"
>
  <div
    className="col-md survey-row"
  >
    <Mutation
      mutation={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "createNewSurvey",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "input",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "input",
                          },
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "createNewSurvey",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "title",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "questions",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "question",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "type",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "directives": Array [],
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "SurveyInput",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "input",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 176,
            "start": 0,
          },
        }
      }
    >
      <Component />
    </Mutation>
  </div>
</styled.div>
`;
